File: app/src/main/java/com/quranapp/android/frags/readerindex/FragReaderIndexChapters.java
package com.quranapp.android.frags.readerindex;

import static androidx.recyclerview.widget.RecyclerView.VERTICAL;

import android.content.Context;

import androidx.recyclerview.widget.GridLayoutManager;

import com.peacedesign.android.utils.WindowUtils;
import com.quranapp.android.adapters.quranIndex.ADPChaptersList;
import com.quranapp.android.views.helper.RecyclerView2;

public class FragReaderIndexChapters extends BaseFragReaderIndex {
    public FragReaderIndexChapters() {
    }

    public static FragReaderIndexChapters newInstance() {
        return new FragReaderIndexChapters();
    }

    @Override
    protected void initList(RecyclerView2 list, Context ctx) {
        super.initList(list, ctx);

        mHandler.post(() -> {
            int spanCount = WindowUtils.isLandscapeMode(ctx) ? 2 : 1;
            GridLayoutManager layoutManager = new GridLayoutManager(ctx, spanCount, VERTICAL, false);
            list.setLayoutManager(layoutManager);
        });

        resetAdapter(list, ctx, false);
    }

    @Override
    protected void resetAdapter(RecyclerView2 list, Context ctx, boolean reverse) {
        super.resetAdapter(list, ctx, reverse);
        ADPChaptersList adapter = new ADPChaptersList(this, ctx, reverse);
        mHandler.post(() -> list.setAdapter(adapter));
    }
}

File: app/src/main/java/com/quranapp/android/frags/readerindex/FragReaderIndexChapters.kt
package com.quranapp.android.frags.readerindex

import android.content.Context
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.peacedesign.android.utils.WindowUtils
import com.quranapp.android.adapters.quranIndex.ADPChaptersList
import com.quranapp.android.views.helper.RecyclerView2

class FragReaderIndexChapters : BaseFragReaderIndex() {
    override fun initList(list: RecyclerView2, ctx: Context) {
        super.initList(list, ctx)
        activity?.runOnUiThread {
            val spanCount = if (WindowUtils.isLandscapeMode(ctx)) 2 else 1
            val layoutManager = GridLayoutManager(ctx, spanCount, RecyclerView.VERTICAL, false)
            list.layoutManager = layoutManager
        }
        resetAdapter(list, ctx, false)
    }

    override fun resetAdapter(list: RecyclerView2, ctx: Context, reverse: Boolean) {
        super.resetAdapter(list, ctx, reverse)
        val adapter = ADPChaptersList(this, ctx, reverse)
        activity?.runOnUiThread {
            list.adapter = adapter
            favChaptersModel.favChapters.observe(viewLifecycleOwner) {
                adapter.onFavChaptersChanged()
            }
        }
    }

    companion object {
        @JvmStatic
        fun newInstance(): FragReaderIndexChapters {
            return FragReaderIndexChapters()
        }
    }
}